tests/args/test_common.py

```diff
@@ -114,3 +113,49 @@ def _(value):
 
     [which] = collect_dumped_objects(result)
     assert which == "re"
+
+
+def test_same_name_for_step_arg_and_example_parameter(pytester):
+    """Test that using the same name for step arg and example parameter works as intended."""
```

Comment:
still not clear what we are testing here. Is it about params being filled from example tables vs from parsing the step name?
If so let's make it clearer in the docstring.

Maybe let's also rewrite the step names, like:
* Given foo gets value "A" by the step parser
* Given bar gets value <example_value> from the example table

This would be much clearer to me

Reply:
I think the other test is actually testing the same thing and this can be closed